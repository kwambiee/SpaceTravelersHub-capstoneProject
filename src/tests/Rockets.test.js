import React from 'react';
import { Provider } from 'react-redux';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import HomePage from '../components/pages/Home';
import '@testing-library/jest-dom/extend-expect';

const RocketsProvider = () => (
  <Provider store={store}>
    <HomePage />
  </Provider>
);

describe('Check Homepage', () => {
  it('renders correctly', () => {
    const component = renderer.create(<RocketsProvider />).toJSON();
    expect(component).toMatchSnapshot();
  });
});

describe('No Rockets Available', () => {
  it('renders correctly', () => {
    render(<RocketsProvider />);
    expect(screen.getByText('No rockets available')).toBeInTheDocument();
  });
});

describe('Rockets Component', () => {
  it('renders rockets', () => {
    const allRockets = renderer.create(<RocketsProvider />).toJSON();
    expect(allRockets).toMatchSnapshot();
  });

  test('Reserve a rocket and cancel reservation', async () => {
    render(<RocketsProvider />);
    const reserveButton = await screen.findAllByText('Reserve Rocket');
    const badges = await screen.findAllByText('Reserved');
    expect(reserveButton[0].textContent).toBe('Reserve Rocket');
    fireEvent.click(reserveButton[0]);
    expect(badges[0].textContent).toBe('Reserved');
    expect(reserveButton[0].textContent).toBe('Cancel Reservation');
  });
});
