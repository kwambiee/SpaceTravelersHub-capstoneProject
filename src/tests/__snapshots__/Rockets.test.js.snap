// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Homepage renders correctly 1`] = `
<ul>
  <li>
    <h1>
      No rockets available
    </h1>
  </li>
</ul>
`;

exports[`Rockets Component renders rockets 1`] = `
<ul>
  <li>
    <h1>
      No rockets available
    </h1>
  </li>
</ul>
`;
